src/ui/public/visualize/loader/embedded_visualize_handler.ts

```diff
@@ -67,6 +69,8 @@ export class EmbeddedVisualizeHandler {
   private readonly appState?: AppState;
   private uiState: PersistedState;
   private dataLoader: VisualizeDataLoader;
+  private dataSubject: Rx.Subject<any>;
+  private data$: Rx.Observable<any>;
```

Comment:
Since we never rewrite this, we usually tend to just make this a `public readonly` variable in other places in Kibana and instead remove the `onVisDataObservable()` function. That way we need less code, and still express the same (and don't end up in that weird naming discussion around method names for observables, since `onXyz` usually is used for registering event handlers, but not simply returning a value, etc.)

Could you please also add some JSDoc then to that variable:

```js
/**
 * This observable will emit every time new data is loaded for the
 * visualization. The emitted value is the loaded data after it has
 * been transformed by the visualization's response handler.
 * This should not be used by any plugin.
 * @ignore
 */
```